### PR TITLE
2931 initialize pk simr in ubuntu

### LINKED ISSUE #2931: Initialize PK-Sim.R in Ubuntu

### PR DESCRIPTION
**Pull request by rwmcintosh**
Fixes #2931

# Description
Some small changes to allow initializing of PKSim.R in Ubuntu

## Type of change

Please mark relevant options with an `x` in the brackets.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Integration tests
- [ ] Unit tests
- [x] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):

# Questions (if appropriate):

### CODE REVIEW

File: appveyor-nightly.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ after_build:
   - cmd: >
       pushd . &&
       cd src\PKSimRDependencyResolution\bin\%CONFIGURATION%\%TARGET_FRAMEWORK% &&
-      "C:\Program Files\7-Zip\7z.exe" a pk-sim-r-dependencies.zip -x@excludedFiles.txt * &&
+      "C:\Program Files\7-Zip\7z.exe" a pk-sim-r-dependencies.zip -x@excludedFiles.txt * -xr!runtimes &&
```
Comment (rwmcintosh) [inline appveyor-nightly.yml:63]: excluding runtimes folder and cherry picking the binaries that are needed for Linux and Windows 64 bit

File: src/PKSim.Assets.Images/PKSim.Assets.Images.csproj [hunk 2]
```diff
@@ -9,6 +9,7 @@
     <InformationalVersion>12.0.0</InformationalVersion>
     <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
     <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
+    <GenerateResourceUsePreserializedResources>true</GenerateResourceUsePreserializedResources>
```
Comment (rwmcintosh) [inline src/PKSim.Assets.Images/PKSim.Assets.Images.csproj:12]: .NET compiler on Ubuntu complained that this was not here. So I added it.

File: src/PKSim.Assets.Images/PKSim.Assets.Images.csproj [hunk 3]
```diff
@@ -28,6 +29,7 @@
   <ItemGroup>
     <PackageReference Include="OSPSuite.Assets" Version="12.0.300" />
     <PackageReference Include="OSPSuite.Assets.Images" Version="12.0.300" />
+    <PackageReference Include="System.Resources.Extensions" Version="8.0.0" />
```
Comment (rwmcintosh) [inline src/PKSim.Assets.Images/PKSim.Assets.Images.csproj:32]: .NET compiler on Ubuntu complained that this was not here. So I added it.

File: src/PKSim.Infrastructure/PKSimConfiguration.cs [hunk 4]
```diff
@@ -69,7 +70,7 @@ private void createDefaultSettingsFolder()
          if (!DirectoryHelper.DirectoryExists(CurrentUserFolderPath))
             DirectoryHelper.CreateDirectory(CurrentUserFolderPath);
 
-         if (!DirectoryHelper.DirectoryExists(AllUsersFolderPath))
+         if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && !DirectoryHelper.DirectoryExists(AllUsersFolderPath))
```
Comment (rwmcintosh) [inline src/PKSim.Infrastructure/PKSimConfiguration.cs:73]: We just don't create the common folder unless it's Windows

File: src/PKSimRDependencyResolution/PKSimRDependencyResolution.csproj [hunk 5]
```diff
@@ -37,4 +37,12 @@
...
     <ProjectReference Include="..\PKSim.R\PKSim.R.csproj" />
   </ItemGroup>
+
+  <Target Name="PostBuild" AfterTargets="PostBuildEvent">
+    <ItemGroup>
+      <AssemblyFiles Include="$(TargetDir)/runtimes/linux-x64/native/netstandard2.0/SQLite.Interop.dll" />
+    </ItemGroup>
+    <Copy SourceFiles="@(AssemblyFiles);" DestinationFolder="$(TargetFolder)" DestinationFiles="@(AssemblyFiles->Replace('/runtimes/linux-x64/native/netstandard2.0/SQLite.Interop.dll',
+		  'libSQLite.Interop.dll'))" />
+  </Target>
```
Comment (rwmcintosh) [inline src/PKSimRDependencyResolution/PKSimRDependencyResolution.csproj:47]: Cross-platform way of copying and renaming files. I want this file to be renamed from SQLite.Interop.dll to libSQLite.Interop.dll and I want it in the build root, not in runtimes folders.

**Review by Yuri05 — APPROVED** [review]

File: src/PKSimRDependencyResolution/PKSimRDependencyResolution.csproj (hunk 5 above)
Comment (Yuri05) [reply src/PKSimRDependencyResolution/PKSimRDependencyResolution.csproj:47]: weird

**Pull request merged**